### PR TITLE
boot script fixes

### PR DESCRIPTION
**Pull request by GalaxyGorilla**
- conform is sensitive to removing the sys.config, it's not removed anymore
- removed a bug with finding the set-cookie flag

### CODE REVIEW

File: priv/rel/files/boot [hunk 1]
```diff
@@ -258,15 +257,15 @@ adjust_name
 
 PIPE_DIR="${PIPE_DIR:-/tmp/erl_pipes/$NAME/}"
 
-# Extract the target cookie
-COOKIE_ARG="$(grep '^-setcookie' "$VMARGS_PATH")"
```
Comment (GalaxyGorilla) [inline priv/rel/files/boot:262]: That gave me an error without any notification.

File: priv/rel/files/boot [hunk 2]
```diff
@@ -139,8 +139,10 @@ generate_vm_args() {
     if [ "$COOKIE_MODE" = "ignore" ]; then
         printf "Cookie will be used from \"$HOME/.erlang.cookie\".\n"
         sed -i -e '/-setcookie/s/^/## removed: /' $VMARGS_PATH
-        COOKIE=`cat $HOME/.erlang.cookie`
-        printf "\n## set due to \"COOKIE_MODE=ignore\"\n-setcookie $COOKIE\n" >> $VMARGS_PATH
+        if [ -s $HOME/.erlang.cookie ] ; then
```
Comment (thz) [inline priv/rel/files/boot]: "ignore" means ignore. we will not read from $HOME, we will not change cookie in vmargs!

File: priv/rel/files/boot [hunk 3]
```diff
@@ -187,10 +189,10 @@ generate_sys_config() {
                 __conform="$ROOTDIR/bin/conform"
             fi
 
-            # remove old sys.config since conform seems to merge this one too
-            rm -v "$GENERATED_CONFIG_DIR/sys.config"
+            # remove old sys.config first from the mutable dir since conform also considers this file for merging
+            rm -v $SYS_CONFIG
```
Comment (thz) [inline priv/rel/files/boot]: we should instead remove $GENERATED_CONFIG_DIR (but only when derived from $RELEASE_MUTABLE_DIR)

File: priv/rel/files/boot [hunk 4]
```diff
@@ -187,10 +189,10 @@ generate_sys_config() {
...
 
-            result="$("$BINDIR/escript" "$__conform" --conf "$__conform_file" --schema "$__schema_file" --config "$SYS_CONFIG" --output-dir "$GENERATED_CONFIG_DIR")"
+            result="$("$BINDIR/escript" "$__conform" --conf "$__conform_file" --schema "$__schema_file" --config "$RELEASE_CONFIG_DIR/sys.config" --output-dir "$GENERATED_CONFIG_DIR")"
```
Comment (thz) [inline priv/rel/files/boot:195]: outch, good catch! is that the reason why sys.config in running-config was considered/merged-in as well?

Comment (GalaxyGorilla) [inline priv/rel/files/boot:195]: yep, I think so!

File: priv/rel/files/boot (hunk 2 above)
Comment (GalaxyGorilla) [inline priv/rel/files/boot]: I think the if statement is redundant and can be removed. It is just there for informational purposes (e.g. a 'complete' vm.args which you can look up).

File: priv/rel/files/boot [hunk 5]
```diff
@@ -115,8 +115,22 @@ adjust_name() {
     esac
 }
 
+clean_mutable_config_dir() {
+    if [ -n "$RELEASE_MUTABLE_DIR" ] ; then
+        printf "cleaning $GENERATED_CONFIG_DIR ..."
+        rm -rfv "$GENERATED_CONFIG_DIR/*"
```
Comment (thz) [inline priv/rel/files/boot]: better remove the *

File: priv/rel/files/boot [hunk 6]
```diff
@@ -115,8 +115,22 @@ adjust_name() {
...
+    fi
+}
+
+remove_mutable_config_dir() {
```
Comment (thz) [inline priv/rel/files/boot:118]: oh wait - do we really need clean and remove?
i see no need for cleaning, when we have removal. (and i also prefer the latter)

File: priv/rel/files/boot [hunk 7]
```diff
@@ -135,19 +148,19 @@ generate_vm_args() {
...
-    
+
     if [ "$COOKIE_MODE" = "ignore" ]; then
-        printf "Cookie will be used from \"$HOME/.erlang.cookie\".\n"
-        sed -i -e '/-setcookie/s/^/## removed: /' $VMARGS_PATH
-        COOKIE=`cat $HOME/.erlang.cookie`
-        printf "\n## set due to \"COOKIE_MODE=ignore\"\n-setcookie $COOKIE\n" >> $VMARGS_PATH
+        printf "No Cookie handling. (\$HOME/.erlang.cookie might apply).\n"
+        #remove the cookie which is set by default from relx
+        sed -i -e 's/^[ \t]*-setcookie/# disabled by exrm-boot: &/' $VMARGS_PATH
```
Comment (thz) [inline priv/rel/files/boot]: we will not fix inadequate release-packaging in our wrapper. ignore means ignore.

File: priv/rel/files/boot [hunk 8]
```diff
@@ -187,10 +200,10 @@ generate_sys_config() {
                 __conform="$ROOTDIR/bin/conform"
             fi
 
-            # remove old sys.config since conform seems to merge this one too
-            rm -v "$GENERATED_CONFIG_DIR/sys.config"
+            # remove old sys.config first from the mutable dir since conform also considers this file for merging
+            rm -fv $SYS_CONFIG
```
Comment (thz) [inline priv/rel/files/boot]: should be obsolete.

File: priv/rel/files/boot [hunk 9]
```diff
@@ -258,15 +271,16 @@ adjust_name
...
 
-# Extract the target cookie
-COOKIE_ARG="$(grep '^-setcookie' "$VMARGS_PATH")"
-if [ -z "$COOKIE_ARG" ]; then
-    echo "vm.args needs to have a -setcookie parameter."
-    exit 1
+if [ ! -z "$(grep '^-setcookie' "$VMARGS_PATH")" ]; then
+    COOKIE_ARG="$(grep '^-setcookie' "$VMARGS_PATH")"
+    COOKIE="$(echo "$COOKIE_ARG" | awk '{print $2}')"
+else
```
Comment (thz) [inline priv/rel/files/boot:266]:
```
if grep -q '^-setcookie' "$VMARGS_PATH" ; then
  COOKIE_ARG=$( grep -m1 '^-setcookie'  "$VMARGS_PATH" )
  COOKIE="${COOKIE_ARG#* }"
fi
```

File: priv/rel/files/boot (hunk 8 above)
Comment (GalaxyGorilla) [inline priv/rel/files/boot]: no, there are cases where you dont want to remove a vm.args, e.g. 'restart'. That's the reason I left this.

Comment (thz) [inline priv/rel/files/boot]: but it should be obsolete with your change in line 204 below.

Comment (GalaxyGorilla) [inline priv/rel/files/boot]: ah thats what you mean

**Pull request merged**